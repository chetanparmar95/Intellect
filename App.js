/**
 * Sample React Native App
 * https://github.com/facebook/react-native
 *
 * @format
 * @flow strict-local
 */

import React, { useState } from 'react';
import {
  SafeAreaView,
  StatusBar,
  StyleSheet,
  Text,
  TouchableOpacity,
  View,
} from 'react-native';
import Ionicons from 'react-native-vector-icons/dist/Ionicons';

import PageProgress from './src/components/PageProgress';
import PyramidScale from './src/components/pyramid_scale';
import MetersScale from './src/components/meters_scale';
import Colors from './src/utils/Colors';

const closeIcon = <Ionicons name="close-outline" size={30} color="#fff" />;

const App = () => {
  const [progress, setProgress] = useState(1) //for maintaining progress of meter scale
  const [level, setLevel] = useState(3) // for maintaining level of pyramid scale
  const [page, setPage] = useState(1) // for maintaining current page indicator
  
  return (
    <SafeAreaView style={styles.backgroundStyle}>
      <StatusBar barStyle={'light-content'} />
      {/* PageProgress indicates current page*/}
      <PageProgress page={page}/>
      <View style={styles.container}>
        {/* For showing current session type */}
        <View style={styles.sessionView}>
          <Text style={styles.sessionText}>RESCUE SESSION: ANGER & FRUSTRATION</Text>
          {closeIcon}
        </View>

        <Text style={styles.headerText}>
          Pick the level your anger & frustration right now
        </Text>
        {/* showing widgets based on the page  */}
        {page == 1 && <MetersScale progress={progress} onProgressChanges = {setProgress}/>}
        {page == 2 && <PyramidScale level={level} setLevel={setLevel}/>}
        {/* Button for navigating between the pages  */}
        <TouchableOpacity testID='Next' activeOpacity={0.8} style={styles.button} onPress={() => setPage(2)}>
          <Text style={styles.buttonText}>{"Next"}</Text>
        </TouchableOpacity>
      </View>
    </SafeAreaView>
  );
};

const styles = StyleSheet.create({
  backgroundStyle: {
    backgroundColor: Colors.primary,
    flex: 1,
  },
  container:{
    marginHorizontal: 10, 
    flex: 1, 
    marginBottom: 5
  },
  sessionView:{
    flexDirection: 'row', 
    alignItems: 'center', 
    justifyContent: 'space-between', 
    marginTop: 10
  },
  sessionText:{
    color: 'white', 
    fontSize: 12
  },
  headerText:{
    color: 'white', 
    fontSize: 18, 
    fontWeight: 'bold', 
    marginTop: 15
  },
  button: {
    alignItems: 'center',
    justifyContent: 'center',
    paddingVertical: 12,
    paddingHorizontal: 32,
    borderRadius: 25,
    elevation: 3,
    backgroundColor: 'white',
  },
  buttonText: {
    fontSize: 16,
    lineHeight: 21,
    fontWeight: 'bold',
    letterSpacing: 0.25,
    color: 'black',
  },
});

export default App;
