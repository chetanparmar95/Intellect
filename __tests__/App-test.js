/**
 * @format
 */

import 'react-native';
import React from 'react';
import App from '../App';

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';
import { render, fireEvent } from '@testing-library/react-native';

it('renders correctly', () => {
  const tree = renderer.create(<App />).toJSON();
  expect(tree).toMatchSnapshot();
});

it("renders meters scale", () => {
  const { queryByTestId } = render(<App />)
  const meters = queryByTestId("meter")
  const pyramid = queryByTestId("pyramid")
  expect(meters).not.toEqual(null)
  expect(pyramid).toEqual(null)
})

it("updates UI when clicking on next button", () => {
  const { queryByTestId, getByTestId, container } = render(<App />)
  fireEvent.press(getByTestId('Next'))
  const pyramid = queryByTestId("pyramid")
  const meters = queryByTestId("meter")
  expect(pyramid).not.toEqual(null)
  expect(meters).toEqual(null)
})
