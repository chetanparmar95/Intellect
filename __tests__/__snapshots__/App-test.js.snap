// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
  style={
    Object {
      "backgroundColor": "#325d6e",
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "flexDirection": "row",
        "height": 4,
        "marginHorizontal": 5,
        "marginTop": 10,
      }
    }
  >
    <View
      style={
        Array [
          Object {
            "borderRadius": 2,
            "flex": 1,
            "height": 4,
            "marginHorizontal": 5,
          },
          Object {
            "backgroundColor": "white",
          },
        ]
      }
    />
    <View
      style={
        Array [
          Object {
            "borderRadius": 2,
            "flex": 1,
            "height": 4,
            "marginHorizontal": 5,
          },
          Object {
            "backgroundColor": "#ffffff55",
          },
        ]
      }
    />
    <View
      style={
        Array [
          Object {
            "borderRadius": 2,
            "flex": 1,
            "height": 4,
            "marginHorizontal": 5,
          },
          Object {
            "backgroundColor": "#ffffff55",
          },
        ]
      }
    />
    <View
      style={
        Array [
          Object {
            "borderRadius": 2,
            "flex": 1,
            "height": 4,
            "marginHorizontal": 5,
          },
          Object {
            "backgroundColor": "#ffffff55",
          },
        ]
      }
    />
    <View
      style={
        Array [
          Object {
            "borderRadius": 2,
            "flex": 1,
            "height": 4,
            "marginHorizontal": 5,
          },
          Object {
            "backgroundColor": "#ffffff55",
          },
        ]
      }
    />
  </View>
  <View
    style={
      Object {
        "flex": 1,
        "marginBottom": 5,
        "marginHorizontal": 10,
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "justifyContent": "space-between",
          "marginTop": 10,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "white",
            "fontSize": 12,
          }
        }
      >
        RESCUE SESSION: ANGER & FRUSTRATION
      </Text>
      <Text
        allowFontScaling={false}
        selectable={false}
        style={
          Array [
            Object {
              "color": "#fff",
              "fontSize": 30,
            },
            undefined,
            Object {
              "fontFamily": "Ionicons",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            Object {},
          ]
        }
      >
        ïˆ§
      </Text>
    </View>
    <Text
      style={
        Object {
          "color": "white",
          "fontSize": 18,
          "fontWeight": "bold",
          "marginTop": 15,
        }
      }
    >
      Pick the level your anger & frustration right now
    </Text>
    <View
      style={
        Object {
          "flex": 1,
          "justifyContent": "space-evenly",
          "marginVertical": 50,
        }
      }
      testID="meter"
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "alignSelf": "center",
            "borderColor": "#FFFFFF55",
            "borderRadius": 300,
            "borderStyle": "dashed",
            "borderWidth": 1,
            "height": 600,
            "justifyContent": "center",
            "width": 600,
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#466d7c",
              "borderRadius": 225,
              "height": 450,
              "justifyContent": "center",
              "position": "absolute",
              "width": 450,
            }
          }
        >
          <Image
            source={
              Object {
                "testUri": "../../../src/assets/images/headspace-progressbar.gif",
              }
            }
            style={
              Object {
                "height": 375,
                "width": 375,
              }
            }
          />
          <Text
            style={
              Object {
                "color": "white",
                "fontSize": 32,
                "fontWeight": "bold",
                "position": "absolute",
              }
            }
            testID="progressText"
          >
            1
          </Text>
          <View
            style={
              Object {
                "position": "absolute",
              }
            }
          >
            <RNSVGSvgView
              bbHeight={468.75}
              bbWidth={468.75}
              focusable={false}
              height={468.75}
              style={
                Array [
                  Object {
                    "backgroundColor": "transparent",
                    "borderWidth": 0,
                  },
                  Object {
                    "flex": 0,
                    "height": 468,
                    "width": 468,
                  },
                ]
              }
              width={468.75}
            >
              <RNSVGGroup>
                <RNSVGGroup
                  matrix={
                    Array [
                      1,
                      0,
                      -0,
                      1,
                      0,
                      0,
                    ]
                  }
                >
                  <RNSVGPath
                    d="M 234.23087944649924 5.00004527680872 A 229.375 229.375 0 1 0 234.375 5"
                    fill={0}
                    propList={
                      Array [
                        "fill",
                        "stroke",
                        "strokeWidth",
                        "strokeLinecap",
                      ]
                    }
                    stroke={0}
                    strokeLinecap={1}
                    strokeWidth={10}
                  />
                </RNSVGGroup>
              </RNSVGGroup>
            </RNSVGSvgView>
          </View>
        </View>
      </View>
      <View
        style={
          Object {
            "alignSelf": "center",
            "width": 600,
          }
        }
      >
        <Slider
          maximumValue={10}
          minimumTrackTintColor="#6edae6"
          minimumValue={0}
          onValueChange={[Function]}
          testID="slider"
          thumbStyle={
            Object {
              "backgroundColor": "#6edae6",
              "borderColor": "#c8f6fc",
              "borderRadius": 10,
              "borderWidth": 3,
              "height": 20,
              "shadowColor": "black",
              "shadowOffset": Object {
                "height": 2,
                "width": 0,
              },
              "shadowOpacity": 0.35,
              "shadowRadius": 2,
              "width": 20,
            }
          }
          trackStyle={
            Object {
              "backgroundColor": "white",
              "borderRadius": 4,
              "height": 10,
              "shadowColor": "black",
              "shadowOffset": Object {
                "height": 1,
                "width": 0,
              },
              "shadowOpacity": 0.15,
              "shadowRadius": 1,
            }
          }
          value={1}
        />
      </View>
    </View>
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "white",
          "borderRadius": 25,
          "elevation": 3,
          "justifyContent": "center",
          "opacity": 1,
          "paddingHorizontal": 32,
          "paddingVertical": 12,
        }
      }
      testID="Next"
    >
      <Text
        style={
          Object {
            "color": "black",
            "fontSize": 16,
            "fontWeight": "bold",
            "letterSpacing": 0.25,
            "lineHeight": 21,
          }
        }
      >
        Next
      </Text>
    </View>
  </View>
</RCTSafeAreaView>
`;
